### ISSUE TITLE
secure access to nest-desktop via access-token as html-parameter

### ISSUE THREAD

**Issue by jhgoebbert** [nest-desktop/nest-desktop#576]
**Is your feature request related to a problem? Please describe.**
We would like to allow nest-desktop to be run on a multi-user system (e.g. login node of an HPC system).
At the moment (to my understanding) anyone who has access to the login node and who knows the port-number which a nest-desktop of any other user  is using, can access nest-desktop and use it.

**Describe the solution you'd like**
It would be fantastic if one can specify a random access-token via cli (or if nest-desktop creates one) on startup (similar to JupyterLab).
eg. `nest-desktop -p 12345 -h localhost -t 9ebdf86eaeff658ab60c671e4ec4431012b155525ab40874`   
This would allow us to start nest-desktop on a multi-user-system with a basic security enabled:  
`https://localhost:12345/index.html?token=9ebdf86eaeff658ab60c671e4ec4431012b155525ab40874`

**Describe alternatives you've considered**
An alternative would be to use UNIX sockets instead of a port but this is less flexible and no option for us at the moment.

**Comment by jhgoebbert** [comment]
More background information:  
We would like to allow to start the `nest-desktop` from within a running JupyterLab session directly from the launcher. This allows us to ensure that authentication & authorization can be outsourced to our jupyterlab service.  
I created a small extension for jupyter-server-proxy which allows just this: https://github.com/jhgoebbert/jupyter-nestdesktop-proxy
![image](https://github.com/nest-desktop/nest-desktop/assets/9266792/255851ab-9da8-41ed-a61f-3011d8e627b6)

This is similar to our extension we use to integrate an html5-enabled remote desktop: https://github.com/FZJ-JSC/jupyter-xprahtml5-proxy But there we include password and encryption support [here](https://github.com/FZJ-JSC/jupyter-xprahtml5-proxy/blob/main/jupyter_xprahtml5_proxy/__init__.py#L119) of which we need at least one for nest-desktop, too.

**Comment by babsey** [comment]
It is resolved with this solution #578.

TL;DR
The frontend doesn't need to be secured, more important is to secure backend.
